### PR TITLE
[chore] re-organize railgun manager

### PR DESCRIPTION
**Pull request by shaneutt**
**What this PR does / why we need it**:

This re-organizes some of the code in railgun, particularly around the controller manager in anticipation of the upcoming beta release.

One of the main features of this change is that manager and controller code is no longer part of our public API, and has been moved to our private API (e.g. we're not supporting linking to the manager/controller libraries from outside).

### CODE REVIEW

File: railgun/internal/cmd/rootcmd/rootcmd.go
Comment (shaneutt) [inline railgun/internal/cmd/rootcmd/rootcmd.go:8]:
```suggestion
	"github.com/spf13/cobra"

	"github.com/kong/kubernetes-ingress-controller/railgun/internal/manager"
```

**Comment by ccfishk** [comment]
Controller/Operator mainly deploy stateful distributed applications - ingress.
What does it mean " this change is that manager and controller code is no longer part of our public API, and has been moved to our private API (e.g. we're not supporting linking to the manager/controller libraries from outside)".  
Our controller does not have any endpoints for user consumption afaik, but we would like to migrate to service module in the long term ?

**Comment by shaneutt** [comment]
> Controller/Operator mainly deploy stateful distributed applications - ingress.
> What does it mean " this change is that manager and controller code is no longer part of our public API, and has been moved to our private API (e.g. we're not supporting linking to the manager/controller libraries from outside)".
> Our controller does not have any endpoints for user consumption afaik, but we would like to migrate to service module in the long term ?

This means that the Golang packages for this code can not be included in packages outside this codebase, the compiler will complain that the packages are `internal`.

**Comment by ccfishk** [comment]
> > Controller/Operator mainly deploy stateful distributed applications - ingress.
> > What does it mean " this change is that manager and controller code is no longer part of our public API, and has been moved to our private API (e.g. we're not supporting linking to the manager/controller libraries from outside)".
> > Our controller does not have any endpoints for user consumption afaik, but we would like to migrate to service module in the long term ?
> 
> This means that the Golang packages for this code can not be included in packages outside this codebase, the compiler will complain that the packages are `internal`.

Confusing - Controller/Operator has not been SDK, API, Service, utility library, a module implement controller pattern, in what case user use the package code ? you mean they refer the way the controller is implemented.

**Comment by shaneutt** [comment]
> > > Controller/Operator mainly deploy stateful distributed applications - ingress.
> > > What does it mean " this change is that manager and controller code is no longer part of our public API, and has been moved to our private API (e.g. we're not supporting linking to the manager/controller libraries from outside)".
> > > Our controller does not have any endpoints for user consumption afaik, but we would like to migrate to service module in the long term ?
> > 
> > 
> > This means that the Golang packages for this code can not be included in packages outside this codebase, the compiler will complain that the packages are `internal`.
> 
> Confusing - Controller/Operator has not been SDK, API, Service, utility library, a module implement controller pattern, in what case user use the package code ? you mean they refer the way the controller is implemented.

Here's an example that I hope will illustrate the problem:

```shell
$ pushd $(mktemp -d)
$ go mod init test
$ cat -> main.go << EOF
package main

import (
	"fmt"

	"github.com/kong/kubernetes-ingress-controller/railgun/pkg/config"
)

func main() {
	_ = config.Config{
		LogLevel: "debug",
	}
	fmt.Println("It works! able to import config.Config from KIC!")
}
EOF
$ go get github.com/kong/kubernetes-ingress-controller@next
$ go mod tidy && go mod verify
$ go run main.go 
It works! able to import config.Config from KIC!
```

After this PR merges it would no longer work:

```shell
$ go run main.go
package command-line-arguments
	main.go:4:2: use of internal package github.com/kong/kubernetes-ingress-controller/railgun/internal/config not allowed
```

Let me know if that helps.

File: railgun/internal/manager/run.go
Comment (mflendrich) [inline railgun/internal/manager/run.go:31]: https://github.com/golang/go/wiki/CodeReviewComments#imports

Please avoid renaming imports unless it resolves a naming conflict.
Prefer renaming packages to aliasing imports. (Discussed out of band)

Comment (shaneutt) [reply railgun/internal/manager/run.go]: I've removed this import alias in the most recent update

File: railgun/Makefile
Comment (mflendrich) [inline railgun/Makefile:5]:
```suggestion
	@rm -rfv testbin/ bin/* coverage.out
```

**Review by mflendrich — APPROVED** [review]

File: railgun/Makefile
Comment (mflendrich) [reply railgun/Makefile:5]: Wait, `-v` conflicts with the `@`.
Please either make this silent or verbose - your call here.

Comment (shaneutt) [reply railgun/Makefile:5]: Personally, I don't mind the verbosity so given the choice I'll keep it, thank you :bow:

**Pull request merged**
